### PR TITLE
fix "graph_output_size" variable

### PR DESCRIPTION
**Pull request by ccclyu**
The variable  *self.graph_output_size* is not defined.

### CODE REVIEW

**Comment by ZhaofengWu** [comment]
Thanks for the PR! Since this variable is never used if `not self.use_semantic_graph`, why do you need it to be outside of the scope of `if self.use_semantic_graph`? Especially since many other things are also only defined within `if self.use_semantic_graph`, e.g., `self.emb_proj` and `self.rgcn`.

**Comment by ccclyu** [comment]
Thanks for your reply.  The proposed PR is to fix the following bug when running your code. 

https://github.com/ZhaofengWu/SIFT/blob/cee0c56123de9719ac7a196858fa14c9690a54a4/models/rgcn_semantic_encoder.py#L87

There may exist a batch whose data are all empty graphs and the code would enter the if statement,  so we need to define the value for "graph_output_size".

File: models/rgcn_semantic_encoder.py [hunk 1]
```diff
@@ -19,10 +19,12 @@ def __init__(self, args):
 
         self.n_graph_attn_composition_layers = args.n_graph_attn_composition_layers
         self.output_size = self.transformer.config.hidden_size
+        
+        self.graph_output_size = args.graph_dim
 
         if self.use_semantic_graph:
             self.emb_proj = nn.Linear(self.transformer.config.hidden_size, args.graph_dim)
-
+
```
Comment (ZhaofengWu) [inline models/rgcn_semantic_encoder.py]: Can you remove this diff?

File: models/rgcn_semantic_encoder.py [hunk 2]
```diff
@@ -19,10 +19,12 @@ def __init__(self, args):
 
         self.n_graph_attn_composition_layers = args.n_graph_attn_composition_layers
         self.output_size = self.transformer.config.hidden_size
+        
+        self.graph_output_size = args.graph_dim
```
Comment (ZhaofengWu) [inline models/rgcn_semantic_encoder.py]: For consistency, can you rename this field to `graph_dim` too, and update `forward`? And also remove the new line above it.

**Review by ZhaofengWu — CHANGES_REQUESTED** [review]
Ahh I see. I misread the code. This was a bug introduced when I refactored the code for publishing. Do you mind making the following changes? I'd happily merge the PR afterward. Thanks!

File: models/rgcn_semantic_encoder.py (hunk 2 above)
Comment (ccclyu) [reply models/rgcn_semantic_encoder.py]: Thanks for your review !  Have changed and committed.

File: models/rgcn_semantic_encoder.py [hunk 3]
```diff
@@ -19,14 +19,15 @@ def __init__(self, args):
 
         self.n_graph_attn_composition_layers = args.n_graph_attn_composition_layers
         self.output_size = self.transformer.config.hidden_size
+        self.graph_dim = args.graph_dim
 
         if self.use_semantic_graph:
-            self.emb_proj = nn.Linear(self.transformer.config.hidden_size, args.graph_dim)
-
+            self.emb_proj = nn.Linear(self.transformer.config.hidden_size, self.graph_dim)
+
```
Comment (ZhaofengWu) [inline models/rgcn_semantic_encoder.py]: Sorry, perhaps I wasn't being clear enough. Do you mind removing the leading spaces here? The rest looks great!

**Review by ZhaofengWu — CHANGES_REQUESTED** [review]

File: models/rgcn_semantic_encoder.py (hunk 3 above)
Comment (ccclyu) [reply models/rgcn_semantic_encoder.py]: Sorry that I still didn't get your point. May I ask which line should be changed?

Comment (ZhaofengWu) [reply models/rgcn_semantic_encoder.py]: If you look at the diff, your change introduced some leading whitespace on the new L26 (the old L25). It's ok if you leave it as is if it's too troublesome to change -- I'm happy to fix that after merging this PR. Thanks for your contribution again!

Comment (ccclyu) [reply models/rgcn_semantic_encoder.py]: Oh, I got it. Would remove them and commit. Thanks for your reminder.

**Pull request merged**